### PR TITLE
OLS-109: Pass conversation history to docs summarizer

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Pass conversation history to docs summarizer

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-109](https://issues.redhat.com//browse/OLS-109)
- Closes #[OLS-109](https://issues.redhat.com//browse/OLS-109)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [x] If it is a core feature, I have added thorough tests.

## Testing
- Unit tests and integration tests. Also can be run via curl/from UI

### CODE REVIEW

File: ols/src/query_helpers/docs_summarizer.py [hunk 1]
```diff
@@ -49,6 +52,7 @@ def summarize(self, conversation: str, query: str, **kwargs) -> tuple[str, str]:
         )
         logger.info(f"{conversation} call settings: {settings_string}")
 
+        # TODO: use history there
```
Comment (onmete) [inline ols/src/query_helpers/docs_summarizer.py:55]: Where it is passed in? Or that is not implemented yet?

Comment (tisnik) [reply ols/src/query_helpers/docs_summarizer.py:55]: in ols query handler

Comment (onmete) [reply ols/src/query_helpers/docs_summarizer.py:55]: I mean here, I don't see arg `history` used in this `summarize` method.

Comment (onmete) [reply ols/src/query_helpers/docs_summarizer.py:55]: I don't understand the benefit of having the work splitter then.
But there is no reason to block this PR though.

Comment (bparees) [reply ols/src/query_helpers/docs_summarizer.py:55]: i'm also not clear on the value of this as a standalone PR but if it's how you prefer to land the work i guess it's ok.

**Pull request merged**